File: algorithms/cpp/169_majority_element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> result;
        for (int i=0;i<nums.size();i++) {
            auto map_it=result.find(nums[i]);
            if (map_it!=result.end())  
            result[nums[i]]+=1;
            else
            result[nums[i]]=1;
            if (result[nums[i]]>nums.size()/2)
            {return nums[i];
            break;}

        }
    }
};

File: algorithms/python3/169_majority_element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash=dict()
        for num in nums:
            if num in hash:
                hash[num]+=1
            else:
                hash[num]=1
            if hash[num]>len(nums)/2:
                return num

a=Solution()

nums=[3,2,3]

print(a.majorityElement(nums))
